shop/src/main/java/ro/msg/learning/shop/controller/ProductServiceController.java

```diff
@@ -0,0 +1,41 @@
+package ro.msg.learning.shop.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import ro.msg.learning.shop.entity.Product;
+import ro.msg.learning.shop.entity.ProductCategory;
+import ro.msg.learning.shop.mapper.ProductDtoMapper;
+import ro.msg.learning.shop.service.ProductService;
+
+import java.util.UUID;
+
+public class ProductServiceController {
+    @Autowired
+    ProductService productService;
+    @Autowired
+    ProductDtoMapper productDtoMapper;
+
+    @RequestMapping(value="/products")
```

Comment:
You have product written in all the mappings that you have. Spring has a solution for this. You can put a @RequestMapping('/product') on the controller itself and it will add it as a prefix before all the function mappings. Thos way you won't repeat yourself ( D.R.Y principle)